avocado/core/varianter.py

```diff
@@ -45,9 +45,30 @@ def generate_variant_id(variant):
     :return: String compounded of ordered node names and a hash of all
              values.
     """
+    def get_variant_name(variant):
+        """
+        To get the variant full name string
+
+        :param variant: Avocado test variant (list of TreeNode-like objects)
+        :return: Complete variant name string
+        """
+        full_name = []
+        for node in variant:
+            var_str = []
+            _parent = node
+            while _parent:
+                var_str.append(_parent.name)
+                _parent = _parent.parent
```

Comment:
why are using a "private" var here?

Reply:
just to differentiate the parent member of the node and local parent variable..

Reply:
So maybe we need to choose better names. I will not block this PR because of this, but IMO would be better to avoid `_` here.

Reply:
I was little hesitant earlier to change the loop variable, but then it is not going cause any trouble, so have removed the additional variable(_parent) now :-)